docs/core_concepts/19_rich_display_rendering/index.md

```diff
@@ -0,0 +1,47 @@
+# Rich Display Rendering
+
+The result renderer will by default render results as a pretty JSON. However,
+there are some cases that are handled specifically, so-called "rich results".
+They are based on the format of the result.
+
+This feature is useful if you want to display an image, a gif or a file instead of the the JSON of it.
+
+If the result is an object/dict with a single key (except for `approval`, which
+needs 3), you can leverage the following rich results:
+
+| Type      | Description                                                 | Example (Deno)                                                                                                       |
+| --------- | ----------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------- |
+| json      | Render the value as a JSON                                  | `return { 'json': { 'a': 1 } }`                                                                                      |
+| table-col | Render the value as a column in a table                     | `return { 'table-col': { 'foo': [42, 8], 'bar': [38, 12] }}`                                                         |
+| table-row | Render the value as a row in a table                        | `return { 'table-row': [ 'foo', 'bar' ]}`                                                                            |
+| html      | Render the value as HTML                                    | `return { 'html': '<div>...</div>' }`                                                                                |
+| png       | Render the value as a PNG image                             | `return { 'png': encode(image) }`                                                                                    |
+| file      | Render an option to download the file                       | `return { 'file': encode(file) }`                                                                                    |
```

Comment:
should mention you can specify filename as additional dict item to set the filename